dolphin-dnn/src/test/java/edu/snu/dolphin/dnn/layers/ConvolutionalLayerTest.java

```diff
@@ -0,0 +1,226 @@
+        .bindImplementation(MatrixFactory.class, MatrixJBLASFactory.class)
+        .build();
+
+    final ConvolutionalLayerConfigurationBuilder builder = ConvolutionalLayerConfigurationBuilder
+        .newConfigurationBuilder()
+        .setInitWeight(0.2f)
+        .setInitBias(0)
+        .setRandomSeed(10)
+        .setKernelHeight(2)
+        .setKernelWidth(2)
+        .setStrideHeight(1)
+        .setStrideWidth(1);
+
+    final ConvolutionalLayerConfigurationBuilder builderWithPadding = ConvolutionalLayerConfigurationBuilder
+        .newConfigurationBuilder()
+        .setInitWeight(0.2f)
+        .setInitBias(0)
+        .setRandomSeed(10)
+        .setPaddingHeight(1)
+        .setPaddingWidth(1)
+        .setKernelHeight(2)
+        .setKernelWidth(2)
+        .setStrideHeight(1)
+        .setStrideWidth(1);
+
+    this.convolutionalLayer =
+        Tang.Factory.getTang().newInjector(layerConf, builder.build())
+        .getInstance(LayerBase.class);
+
+    this.convolutionalWithPaddingLayer =
+        Tang.Factory.getTang().newInjector(layerConf, builderWithPadding.build())
+        .getInstance(LayerBase.class);
+  }
+
+  @Test
+  public void testConvolutionalActivation() {
+    final Matrix convolutionalActivation = convolutionalLayer.feedForward(input);
+    assertTrue(expectedConvolutionalActivation.compare(convolutionalActivation, TOLERANCE));
+  }
+
+  @Test
+  public void testConvolutionalBackPropagate() {
+    final Matrix error = convolutionalLayer.backPropagate(input, expectedConvolutionalActivation, nextError);
+    assertTrue(expectedConvolutionalError.compare(error, TOLERANCE));
+  }
+
+  @Test
+  public void testConvolutionalWithPaddingActivation() {
+    final Matrix convolutionalActivation = convolutionalWithPaddingLayer.feedForward(input);
+    assertTrue(expectedConvolutionalWithPaddingActivation.compare(convolutionalActivation, TOLERANCE));
+  }
+
+  @Test
+  public void testConvolutionalWithPaddingBackPropagate() {
+    final Matrix error = convolutionalWithPaddingLayer
+        .backPropagate(input, expectedConvolutionalWithPaddingActivation, nextErrorWithPadding);
+    assertTrue(expectedConvolutionalWithPaddingError.compare(error, TOLERANCE));
+  }
+
+  @Test
+  public void testConvolutionalGradient() {
+    final LayerParameter convolutionalLayerParams = convolutionalLayer.generateParameterGradient(input, nextError);
+    assertTrue(expectedConvolutionalLayerParams
+        .getBiasParam().compare(convolutionalLayerParams.getBiasParam(), TOLERANCE));
+    assertTrue(expectedConvolutionalLayerParams
+        .getWeightParam().compare(convolutionalLayerParams.getWeightParam(), TOLERANCE));
+  }
+
+  @Test
+  public void testConvolutionalWithPaddingGradient() {
+    final LayerParameter convolutionalLayerParams =
+        convolutionalWithPaddingLayer.generateParameterGradient(input, nextErrorWithPadding);
+    assertTrue(expectedConvolutionalLayerWithPaddingParams
```

Comment:
Checking the equality of two layer parameters is used several times. Why don't we create a method for this?